python/graphstorm/model/edge_decoder.py

```diff
@@ -713,6 +714,45 @@ def out_dims(self):
         """
         return 1
 
+class LinkPredictContrastiveDotDecoder(LinkPredictDotDecoder):
```

Comment:
once we get to a better solution, this class is no longer needed?

Reply:
if so, we shouldn't expose this API to users. in the docstring, we should say that this is a temporary solution.

Reply:
> once we get to a better solution, this class is no longer needed?

I think so. But it may take time for us to implement a better solution.
We can hide it from our users.

Reply:
Add a TODO.